### DISCUSSION TITLE
Is there a workaround to groupby nested relations?

### DISCUSSION THREAD

**Discussion by gmcamposano** [prisma/prisma#16255] [Q&A]
Firstly, I had to structure it this way, as i couldn't find a way to group "Factura" table with the month date of created_at field. 
Example: I want to get the total of sales per month, so the only way I could find to do this is to include a month and year row. (if this is possible, please let me know)

So it happens now that I included it in Numero_de_factura table and I am trying to group by month and also add the total of sales in that month. Is this possible as I cannot find a way to use relations with aggregation function?

```
model Numero_de_factura {
id      String @id @default(uuid()) 
month   Int @default(10) 
year    Int @default(2022) 
factura Factura[] 
}

model Factura { 
id                   String @id @default(uuid()) 
total                Int 
producto_id          String 
numero_de_factura_id String 
numero_de_factura    Numero_de_factura @relation(fields: [numero_de_factura_id], references: [id]) 
created_at           DateTime @default(now()) 
updated_at           DateTime @updatedAt 
}
```

What I was trying to do and did not work was something similar to:

```
const facturas = await prisma.numero_de_factura.groupBy({
    by: ['month'],
    include: {
      factura: true
    },
    _sum: {
      total: true
    }
  })
```

**Comment by nurul3101** [comment]
Hello @gmcamposano 👋 

Grouping by fields of relation model is not supported yet. We have a Feature Request for it here: [#8744](https://github.com/prisma/prisma/issues/8744). Also we have a feature request for native support of GroupBy and aggregation over date ranges which you are looking for: #6653. If you could leave a 👍 to these feature requests then it would help in prioritising it.
For now you would need to use [raw query](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access).

**Reply by vladi-stevanovic** [reply]
👋 Hi @gmcamposano , did you perhaps have a chance to check Nurul's recommendation? Let us know if you still have any questions! 

Or, if you've found a solution, we'd love to know how you solved this problem. Feel free to post it in this thread and "Mark as answer" if this discussion is now solved!

**Reply by gmcamposano** [reply]
@vladi-stevanovic Marking this as answered would not be correct due to the fact that is a feature request, and it can mislead others that have the same question.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
